src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -440,12 +440,17 @@ private bool TryGetPropertyForInputModelProperty(Func<ObjectTypeProperty, bool>
                 discriminatorPropertyName = parent.Discriminator.SerializedName;
                 property = parent.Discriminator.Property;
             }
-            else
+            else if (_derivedTypes != null)
```

Comment:
lets let the null ref happen like before IF it ever does so that its obvious where the issue is